internal/evaluation_target/application_snapshot_image/application_snapshot_image.go

```diff
@@ -169,6 +169,9 @@ func (a *ApplicationSnapshotImage) ValidateImageSignature(ctx context.Context) e
 
 // ValidateAttestationSignature executes the cosign.VerifyImageAttestations method
 func (a *ApplicationSnapshotImage) ValidateAttestationSignature(ctx context.Context) error {
+	// TODO: This is a hack of course. We need maybe not set ClaimVerifier, and make a copy
+	// of checkOpts so we can set the correct one for signature vs attestation.
+	a.checkOpts.ClaimVerifier = cosign.IntotoSubjectClaimVerifier
```

Comment:
Derived from the policy? So that the CheckOpts created in `policy.CheckOpts()` is already preconfigured?

Reply:
Yeah. Updated with the copy approach described in the comment.